From 203b2e4609fec8ae9681ecae41a7d58c4ab6f04c Mon Sep 17 00:00:00 2001
From: "johan.bohlin" <johan.bohlin@intel.com>
Date: Fri, 13 Feb 2015 14:04:48 +0100
Subject: [PATCH] ASoC: Intel: Add a check for dfw loading

Only load dfw binary if dfw is enabled.

Issue: GMINL-3679

Change-Id: I89c4738d5f9bd19efb87fa662677fe11d3578749
Signed-off-by: johan.bohlin <johan.bohlin@intel.com>
---
 sound/soc/intel/pcm.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/pcm.c b/sound/soc/intel/pcm.c
index ac3a87b..4c38901 100644
--- a/sound/soc/intel/pcm.c
+++ b/sound/soc/intel/pcm.c
@@ -947,11 +947,14 @@ static int sst_platform_probe(struct platform_device *pdev)
 	struct file *file;
 
 	pr_debug("sst_platform_probe called\n");
-	file = filp_open("/etc/firmware/dfw_sst.bin", O_RDONLY, 0);
 
-	if (IS_ERR(file)) {
-		pr_info("sst_platform_probe is deferred\n");
-		return -EPROBE_DEFER;
+	if(pdata->dfw_enable) {
+		file = filp_open("/etc/firmware/dfw_sst.bin", O_RDONLY, 0);
+
+		if (IS_ERR(file)) {
+			pr_info("sst_platform_probe is deferred\n");
+			return -EPROBE_DEFER;
+		}
 	}
 
 	sst = devm_kzalloc(&pdev->dev, sizeof(*sst), GFP_KERNEL);
-- 
1.9.1

